workflows/differentialabundance.nf

```diff
@@ -568,6 +712,67 @@ workflow DIFFERENTIALABUNDANCE {
         .mix(PLOT_EXPLORATORY.out.versions)
         .mix(PLOT_DIFFERENTIAL.out.versions)
 
+    // ========================================================================
+    // ShinyNGS app
+    // ========================================================================
+
+    ch_shinyngs = ch_paramsets
+        .filter{ it.params.shinyngs_build_app }
+
+    // to prepare the input for shinyngs app we need to first make sure that the differential
+    // results are parsed in the same order as the contrast file
+    // To do so, as we cannot rely on the order of the channels as they are asynchronous, we
+    // create temporary contrast files with the entries based on the order of the gathered
+    // differential results
+
+    // create a channel with the differential results and the corresponding map with
+    // the contrast entries
+    ch_differential_with_contrast = ch_shinyngs
+        .join( ch_differential_results.groupTuple() )   // [meta, [meta_full], [differential results]]
+        .join( ch_contrasts )                           // [meta, [contrast], [variable], [reference], [target], [formula], [comparison]]
+        .map { meta, meta_full, results, contrast, variable, reference, target, formula, comparison ->
+            // extract the contrast entries from the meta dynamically
+            // in this way we don't need to harcode the contrast keys
+            def contrast_keys = contrast[0].keySet()
+            def contrast_maps = meta_full.collect { it.subMap(contrast_keys) }
+            [meta, results, contrast_maps]   // [meta, [differential results], [contrast maps]]
+        }
+
+    // save temporary contrast csv files with the entries ordered by the differential results
```

Comment:
Can you do me a favour and capitalise the start of all your comment sentences? The lower case is making me twitch ;-)